### PR TITLE
Add tcp support

### PR DESCRIPTION
**Pull request by lu-zero**

### CODE REVIEW

File: srt-transmit/src/helptext.txt [hunk 1]
```diff
@@ -1,78 +1,88 @@
 Supported protocols:
 
-UDP - send over a UDP port. 
+ TCP - send over a TCP connection
+    example:
+        srt-transmit
+            tcp://host:1234 \
+                # ^- connect to host:1234 \
+            tcp://host:1234?listen
```
Comment (russelltg) [inline srt-transmit/src/helptext.txt:8]: This is not how I've exposed it for SRT (I wanted to match the reference impl, although i'm not sure what they do for TCP), what we do is the host part of the URL is always a remote host, so if left out it's assumed that you want to bind to a local port. I added the `interface` query parameter to be able to specify what local interface to bind to (if not 0.0.0.0)

**Review by russelltg — CHANGES_REQUESTED** [review]
Overall looks good, although you should add a unit test for it in stransmit_cmdline.rs

File: srt-transmit/src/helptext.txt (hunk 1 above)
Comment (lu-zero) [reply srt-transmit/src/helptext.txt:8]: that I know the `srt-*-transmit` support srt, udp and file.

```
Supported schemes:
	srt: use HOST, PORT, and PARAM for setting socket options
	udp: use HOST, PORT and PARAM for some UDP specific settings
	file: only as file://con for using stdin or stdout
```

I'll change the scheme handling to match what you use for the other protocols

Comment (lu-zero) [reply srt-transmit/src/helptext.txt:8]: Done.

**Review by russelltg — CHANGES_REQUESTED** [review]
Overall very close to what I want, thanks a lot. Hopefully those failing tests is a flake, we'll see. If it isn't and you don't have a windows machine to test on I can debug those.

File: srt-transmit/src/main.rs [hunk 2]
```diff
@@ -427,6 +455,30 @@ fn resolve_output(output_url: DataType) -> Result<SinkStream, Error> {
                         once(make_srt_ouput(output_addr, output_url, output_local_port)).boxed()
                     }
                 }
+                "tcp" => {
+                    if let Some(output) = output_addr {
+                        once(async move {
+                            let stream = TcpStream::connect(output).await?;
+                            Ok(Framed::new(stream, BytesCodec::new())
```
[the remark this answers is not in the document]
Comment (lu-zero) [reply srt-transmit/src/main.rs:462]: I'll prepare a commit, sure :)

File: srt-transmit/src/helptext.txt [hunk 3]
```diff
@@ -1,78 +1,88 @@
...
+                # ^- bind to interface host:1234 and listen for data
+    Settings:
```
[the remark this answers is not in the document]
Comment (lu-zero) [reply srt-transmit/src/helptext.txt:10]: Looks like a rebase lost the update, I'll fix that :)

File: srt-transmit/tests/stransmit_cmdline.rs [hunk 4]
```diff
@@ -183,6 +183,17 @@ mod stransmit_rs_snd_rcv {
         .await
     }
 
+    #[tokio::test]
+    async fn basic_tcp() -> Result<(), Error> {
+        test_send(
+            3000,
+            &["tcp://:3001", "udp://127.0.0.1:3002"],
+            &["udp://:3000", "tcp://127.0.0.1:3001"],
```
[the remark this answers is not in the document]
Comment (lu-zero) [reply srt-transmit/tests/stransmit_cmdline.rs:191]: I'd need to 

> Overall very close to what I want, thanks a lot. Hopefully those failing tests is a flake, we'll see. If it isn't and you don't have a windows machine to test on I can debug those.

Sadly I do not.

**Comment by russelltg** [comment]
What's the motivation of the move to `Command`?

Also, it seems like the tests are failing because TCP is obviously streambased, so the tests should be updated to not require specific packet boundaries, but instead just a specific list of data.

**Comment by lu-zero** [comment]
> What's the motivation of the move to `Command`?

While trying to figure out why the tests were failing randomly I decided to switch to tokio::Command to keep everything on the async side. I can skip it if you prefer.

> 
> Also, it seems like the tests are failing because TCP is obviously streambased, so the tests should be updated to not require specific packet boundaries, but instead just a specific list of data.
> 

Strangely enough it never happens on my linux and macos system, I'll update it now.

> Failing on windows because access denied.....fantastic.

That's surprising.

**Comment by russelltg** [comment]
Failing on windows.....I'll take a look at it in a bit.

**Comment by lu-zero** [comment]
Could you get a backtrace on windows? I wonder if there is a `Map(Result::unwrap)` that should be a `take_while(Result::is_ok)`

**Comment by russelltg** [comment]
I'm just going to disable the test on windows--I'm imagining that any regression would come up on linux or macos, and I like this feature and don't wanna dig too hard on it.

**Pull request merged**